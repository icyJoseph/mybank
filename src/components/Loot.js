import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchBitcoin } from "../actions/bitcoin";

export class Loot extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
  }

  calculateBitcoinValue = (bitcoin, balance) => {
    return bitcoin.bpi
      ? balance / parseInt(bitcoin.bpi.USD.rate.replace(",", ""), 10)
      : "";
  };

  render() {
    const { balance, bitcoin } = this.props;
    return (
      <div>
        <h3>Bitcoin balance: {this.calculateBitcoinValue(bitcoin, balance)}</h3>
        <p>
          Powered By{" "}
          <a
            href="https://www.coindesk.com/price"
            target="_blank"
            rel="noopener noreferrer"
          >
            CoinDesk
          </a>
        </p>
      </div>
    );
  }
}

export default connect(
  state => ({
    balance: state.balance,
    bitcoin: state.bitcoin
  }),
  { fetchBitcoin }
)(Loot);
