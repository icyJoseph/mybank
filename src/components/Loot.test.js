import React from "react";
import { mount, shallow } from "enzyme";
import { Loot } from "./Loot";

describe("Loot", () => {
  const mockfetchBitcoin = jest.fn();
  const props = { balance: 10, bitcoin: {} };
  let loot = shallow(<Loot {...props} />);

  it("renders properly", () => {
    expect(loot).toMatchSnapshot();
  });

  describe("when mounted", () => {
    beforeEach(() => {
      props.fetchBitcoin = mockfetchBitcoin;
      loot = mount(<Loot {...props} />);
    });

    it("dispatched the fetchBitcoint() it receives from props", () => {
      expect(mockfetchBitcoin).toHaveBeenCalled();
    });
  });
});
