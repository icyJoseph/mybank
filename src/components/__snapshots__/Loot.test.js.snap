// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loot
    balance={10}
    bitcoin={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Powered By CoinDesk
</h3>,
        <h3>
          Bitcoin balance: 
          Object {}
</h3>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Powered By CoinDesk",
        },
        "ref": null,
        "rendered": "Powered By CoinDesk",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Bitcoin balance: ",
            Object {},
          ],
        },
        "ref": null,
        "rendered": Array [
          "Bitcoin balance: ",
          Object {},
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Powered By CoinDesk
</h3>,
          <h3>
            Bitcoin balance: 
            Object {}
</h3>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Powered By CoinDesk",
          },
          "ref": null,
          "rendered": "Powered By CoinDesk",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Bitcoin balance: ",
              Object {},
            ],
          },
          "ref": null,
          "rendered": Array [
            "Bitcoin balance: ",
            Object {},
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
